### PR TITLE
Cast interface events

### PR DESCRIPTION
**Pull request by Boda805**
## Solution

Solves #594

### CODE REVIEW

**Comment by Boda805** [comment]
Had some oddity with f97aa47, git said I had changes to foundryup/install somehow, but file looked the same and I couldn't discard, so I tried to stop tracking.

**Comment by mds1** [comment]
Tested on two contracts and works great! 👌 Neither had any ABIEncoderV2 params, as I'm not sure of contracts offhand to test that with though

File: utils/src/lib.rs [hunk 1]
```diff
@@ -636,6 +636,49 @@ fn format_param(param: &Param, structs: &mut HashSet<String>) -> String {
     }
 }
 
+fn format_event_params(param: &EventParam, structs: &mut HashSet<String>) -> String {
+    let (kind, v2_struct) = match param.kind {
```
Comment (gakonst) [inline utils/src/lib.rs:640]: Can you refactor this to be reused across both functions and events to avoid code duplication?

**Review by gakonst — CHANGES_REQUESTED** [review]
LGTM, small refactor needed, and if you can also add to the tests

**Comment by Boda805** [comment]
Thanks for helping get finished up, I was working on but would've taken bit longer. One nit I was going to fix was the 's' should probably be dropped off of format_event_params.

**Pull request merged**